api/apps/api/src/modules/projects-acl/project-acl.service.ts

```diff
@@ -5,26 +5,91 @@ import { intersection } from 'lodash';
 import { Roles } from '@marxan-api/modules/users/role.api.entity';
 import { Injectable } from '@nestjs/common';
+import {
+  AccessControlService,
+  Permit,
+} from '../access-control/access-control-service.interface';
 
 /**
  * Debt: neither UsersProjectsApiEntity should belong to projects
  * nor the Roles should belong to users
  */
 @Injectable()
-export class ProjectAclService {
-  private readonly canPublishRoles = [
-    Roles.project_admin,
+export class ProjectAclService extends AccessControlService {
+  private readonly canCreateScenarioRoles = [Roles.project_owner];
+  private readonly canEditScenarioRoles = [Roles.project_owner];
+  private readonly canViewSolutionRoles = [Roles.project_owner];
+  private readonly canPublishProjectRoles = [Roles.project_owner];
+  private readonly canViewProjectRoles = [
     Roles.project_owner,
-    Roles.organization_admin,
-    Roles.organization_owner,
+    Roles.project_contributor,
+    Roles.project_viewer,
   ];
 
   constructor(
     @InjectRepository(UsersProjectsApiEntity)
     private readonly roles: Repository<UsersProjectsApiEntity>,
-  ) {}
+  ) {
+    super();
+  }
+  // TODO: this will be changed in the following release of user requirements.
+  // For now, anyone should be able to create projects, regardless of having roles or not.
+  // In the future project creation will be limited to organization contributors, so this logic will be moved to the access control module
+  async canCreateProject(_userId: string, _projectId: string): Promise<Permit> {
+    return true;
+  }
+
+  async canViewProject(userId: string, projectId: string): Promise<Permit> {
+    const roles = (
+      await this.roles.find({
+        where: {
+          projectId,
+          userId,
+        },
+      })
+    ).flatMap((role) => role.roleName);
+
+    return intersection(roles, this.canViewProjectRoles).length > 0;
+  }
+
+  async canPublishProject(userId: string, projectId: string): Promise<Permit> {
+    const roles = (
+      await this.roles.find({
+        where: {
+          projectId,
+          userId,
+        },
+      })
+    ).flatMap((role) => role.roleName);
 
-  async canPublish(userId: string, projectId: string): Promise<boolean> {
+    return intersection(roles, this.canPublishProjectRoles).length > 0;
+  }
+
+  async canCreateScenario(userId: string, projectId: string): Promise<Permit> {
+    const roles = (
+      await this.roles.find({
+        where: {
+          projectId,
+          userId,
+        },
+      })
+    ).flatMap((role) => role.roleName);
+
+    return intersection(roles, this.canCreateScenarioRoles).length > 0;
+  }
+  async canEditScenario(userId: string, projectId: string): Promise<Permit> {
+    const roles = (
+      await this.roles.find({
+        where: {
+          projectId,
+          userId,
+        },
+      })
+    ).flatMap((role) => role.roleName);
```

Comment:
Another minor: could be a good fit to extract this call to standalone private function like `getUserRolesWithinProject(...)` to avoid repetition.